core/controlplane/config/templates/stack-template.json

```diff
@@ -692,25 +696,11 @@
           {{$sgRef}}
           {{end}}
         ],
-        "PlacementTenancy": "{{$.Etcd.Tenancy}}",
-        "UserData": { "Fn::Base64": { "Fn::Join" : ["\n", [
-          "#!/bin/bash -xe",
-          {"Fn::Join":["",[ "echo '{{$.StackNameEnvVarName}}=", { "Ref": "AWS::StackName" }, "' >> {{$.EtcdNodeEnvFileName}}" ]]},
-          "echo '{{$.EtcdIndexEnvVarName}}={{$etcdIndex}}' >> {{$.EtcdNodeEnvFileName}}",
-          " . /etc/environment",
-          "export COREOS_PRIVATE_IPV4 COREOS_PRIVATE_IPV6 COREOS_PUBLIC_IPV4 COREOS_PUBLIC_IPV6",
-          "REGION=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | jq -r '.region')",
-          "USERDATA_FILE={{$.UserDataEtcdFileName}}",
-          "while ! /usr/bin/rkt run \\",
-          "   --net=host \\",
-          "   --volume=dns,kind=host,source=/etc/resolv.conf,readOnly=true --mount volume=dns,target=/etc/resolv.conf  \\",
-          "   --volume=awsenv,kind=host,source=/var/run/coreos,readOnly=false --mount volume=awsenv,target=/var/run/coreos \\",
-          "   --trust-keys-from-https \\",
-          "   {{$.AWSCliImage.Options}}{{$.AWSCliImage.RktRepo}} --exec=aws -- s3 --region $REGION  cp {{ $.UserDataEtcdS3URI }} /var/run/coreos/$USERDATA_FILE; do",
-          "sleep 1",
-          "done",
-          "exec /usr/bin/coreos-cloudinit --from-file /var/run/coreos/$USERDATA_FILE"
-        ]]}}
+        "PlacementTenancy": "{{$.Etcd.Tenancy}}"
+        {{- if $.UserDataEtcd.Parts.instance }}
+        ,
+          "UserData": {{ $.UserDataEtcd.Parts.instance.Base64 true (dict "etcdIndex" $etcdIndex) | checkSizeLessThan 16384 | quote }}
+        {{- end}}
```

Comment:
Suppose we must always have the instance part of user-data, could you omit this `if-else` block like:

```
"PlacementTenancy": "{{$.Etcd.Tenancy}}",
"UserData": {{ $.UserDataEtcd.Parts.instance.Base64 true (dict "etcdIndex" $etcdIndex) | checkSizeLessThan 16384 | quote }}
```

so that the UserData in the stack template won't be accidentally emptied when we broke cloud-config-etcd?